### ISSUE TITLE
WSOD Route error after build

### ISSUE THREAD

**Issue by jenter** [acquia/headless-lightning#91]
I have created a new project using BLT & the Headless Lightning distro. 
There seems to be an outstanding bug which is causing issues post installation. 
The issues seems to stem from a mismatch with the an entity and the media module (or sub-dependency). 

**WSOD Error (affects all admin pages)**
```
The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">Symfony\Component\Routing\Exception\RouteNotFoundException</em>: Route &quot;entity.media_bundle.collection&quot; does not exist. in <em class="placeholder">Drupal\Core\Routing\RouteProvider-&gt;getRouteByName()</em> (line <em class="placeholder">190</em> of <em class="placeholder">core/lib/Drupal/Core/Routing/RouteProvider.php</em>). <pre class="backtrace">Drupal\Core\Routing\UrlGenerator-&gt;getRoute(&#039;entity.media_bundle.collection&#039;) (Line: 130)
Drupal\Core\Routing\UrlGenerator-&gt;getPathFromRoute(&#039;entity.media_bundle.collection&#039;, Array) (Line: 68)
Drupal\Core\Render\MetadataBubblingUrlGenerator-&gt;getPathFromRoute(&#039;entity.media_bundle.collection&#039;, Array) (Line: 790)
Drupal\Core\Url-&gt;getInternalPath() (Line: 135)
Drupal\Core\Utility\LinkGenerator-&gt;generate(&#039;Media Types&#039;, Object) (Line: 94)
Drupal\Core\Render\Element\Link::preRenderLink(Array)
call_user_func(Array, Array) (Line: 378)
Drupal\Core\Render\Renderer-&gt;doRender(Array, ) (Line: 195)
Drupal\Core\Render\Renderer-&gt;render(Array) (Line: 490)
Drupal\Core\Template\TwigExtension-&gt;escapeFilter(Object, Array, &#039;html&#039;, NULL, 1) (Line: 105)
__TwigTemplate_141c9992e95049b83c321703ebb80b76fd76e9689163b934f585c3528030a359-&gt;getmenu_links(Array, Object, 0) (Line: 48)
__TwigTemplate_141c9992e95049b83c321703ebb80b76fd76e9689163b934f585c3528030a359-&gt;doDisplay(Array, Array) (Line: 432)
Twig_Template-&gt;displayWithErrorHandling(Array, Array) (Line: 403)
Twig_Template-&gt;display(Array) (Line: 411)
Twig_Template-&gt;render(Array) (Line: 64)
twig_render_template(&#039;core/themes/stable/templates/navigation/menu--toolbar.html.twig&#039;, Array) (Line: 384)
Drupal\Core\Theme\ThemeManager-&gt;render(&#039;menu__toolbar&#039;, Array) (Line: 437)
Drupal\Core\Render\Renderer-&gt;doRender(Array, 1) (Line: 195)
Drupal\Core\Render\Renderer-&gt;render(Array, 1) (Line: 151)
Drupal\Core\Render\Renderer-&gt;Drupal\Core\Render\{closure}() (Line: 582)
Drupal\Core\Render\Renderer-&gt;executeInRenderContext(Object, Object) (Line: 152)
Drupal\Core\Render\Renderer-&gt;renderPlain(Array) (Line: 339)
_toolbar_do_get_rendered_subtrees(Array)
call_user_func(&#039;_toolbar_do_get_rendered_subtrees&#039;, Array) (Line: 378)
Drupal\Core\Render\Renderer-&gt;doRender(Array, 1) (Line: 195)
Drupal\Core\Render\Renderer-&gt;render(Array, 1) (Line: 151)
Drupal\Core\Render\Renderer-&gt;Drupal\Core\Render\{closure}() (Line: 582)
Drupal\Core\Render\Renderer-&gt;executeInRenderContext(Object, Object) (Line: 152)
Drupal\Core\Render\Renderer-&gt;renderPlain(Array) (Line: 309)
toolbar_get_rendered_subtrees() (Line: 367)
_toolbar_get_subtrees_hash() (Line: 167)
toolbar_toolbar()
call_user_func_array(&#039;toolbar_toolbar&#039;, Array) (Line: 402)
Drupal\Core\Extension\ModuleHandler-&gt;invokeAll(&#039;toolbar&#039;) (Line: 81)
Drupal\toolbar\Element\Toolbar::preRenderToolbar(Array)
call_user_func(Array, Array) (Line: 378)
Drupal\Core\Render\Renderer-&gt;doRender(Array) (Line: 450)
Drupal\Core\Render\Renderer-&gt;doRender(Array, ) (Line: 195)
Drupal\Core\Render\Renderer-&gt;render(Array) (Line: 490)
Drupal\Core\Template\TwigExtension-&gt;escapeFilter(Object, Array, &#039;html&#039;, NULL, 1) (Line: 86)
__TwigTemplate_be8c7bbb9c824f2826368d7c8da984c6279779db72a67fd8056a00bb23b816f2-&gt;doDisplay(Array, Array) (Line: 432)
.................
```

**Comment by danepowell** [comment]
I can reproduce this by following these instructions: https://dev.acquia.com/blog/creating-a-decoupled-drupal-application-in-30-minutes-with-lightning-blt-and-drupalvm/28/11/2017/18886

The error occurs as soon as you log in for the first time, and is not mitigated by a cache clear or any other workaround I can find.

No clue as to the cause.

**Comment by jenter** [comment]
thanks @danepowell, those are the same steps I was following. 

cc @balsama

**Comment by phenaproxima** [comment]
Okay, I'm going to take a look at this today because @balsama is out most of this week. I'll keep everyone posted here.

**Comment by jenter** [comment]
awesome, appreciate it @phenaproxima.

**Comment by phenaproxima** [comment]
I have a theory: the tutorial instructions are requiring too old a version of Headless Lightning. (The tutorial uses a reasonably loose constraint, but Composer is resolving the dependency to an old version that doesn't work anymore.)

I followed the steps to the letter, and found this very incriminating code in the version of content_model_ui.module installed by BLT:

```
  $local_tasks['entity.media_bundle.collection']['title'] = t('Media Types');
  $local_tasks['entity.media_bundle.collection']['base_route'] = 'admin.content_models';
  $local_tasks['entity.media_bundle.collection']['weight'] = 2;
```

HEAD of Headless Lightning does not have this, because as the exception helpfully points out, there *is* no more entity.media_bundle.collection route. It went away when Lightning switched to the core Media module. Instead, it has this, which is correct:

```
  $local_tasks['entity.media_type.collection']['title'] = t('Media Types');
  $local_tasks['entity.media_type.collection']['base_route'] = 'admin.content_models';
  $local_tasks['entity.media_type.collection']['weight'] = 2;
```

So I believe that BLT needs to be updated to bring in a newer version of Headless Lightning. Going backwards, here are the tags which have the correct code:

* 1.2.0-alpha2
* 1.2.0-alpha1
* 1.1.0-alpha3
* 1.1.0-alpha2

BLT, for me, defaulted to 1.1.0-alpha1, which is wrong. So I think that may be what's going on here.

Thoughts?

**Comment by jenter** [comment]
@danepowell @grasmash can either of you comment on the proper version of BLT to pin?

cc @phenaproxima

**Comment by phenaproxima** [comment]
Any version using 1.1.0-alpha2 or higher of Headless Lightning should do the trick, I think.

**Comment by danepowell** [comment]
Oh duh, I should have checked if it was pulling the latest Headless Lightning version. Sorry about that.

Note that this has nothing to do with BLT. You `composer require` Headless Lightning as part of the tutorial.

So the solution here is to ping the docs team and get them to update the tutorial to require Headless Lightning 1.2.0-alpha2. I'll do that.

**Comment by jenter** [comment]
I confirming testing steps on my end with following steps. 

[X] composer create-project --no-interaction acquia/blt-project headless
[X] updated composer to following:
```
{
    "name": "acquia/blt-project",
    "description": "Project template for Drupal 8 sites built with BLT.",
    "type": "project",
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "acquia/blt": "^8.9.12",
        "acquia/lightning": "^3.0.0",
        "acquia/headless_lightning": "^1.1.0-alpha2",
        "geerlingguy/drupal-vm": "~4.3"
    },
    "require-dev": {},
    .......
```
[X] blt vm
[X] blt setup

Everything is working proper now! ...thanks for your input @phenaproxima @danepowell @grasmash

**Comment by balsama** [comment]
Just catching up. Looks like we can close this now?

I think the reason it originally resolved to the 1.1.x branch of Headless is that BLT requires the 2.2.x branch of Lightning... and the 1.2.x branch of Headless needs the 3.0.x branch of Lightning.

Anyway, we should get BLT updated to the 3.0.x branch soon so we don't see more of this.

**Comment by danepowell** [comment]
The upcoming BLT 9.x release depends on Lightning 3: https://github.com/acquia/blt/blob/08dd840bc7f40e090d600bc083b1f7683555b002/composer.suggested.json#L9

It's currently on alpha4, hopefully a stable release is imminent.
